python/cudf_polars/tests/experimental/test_join.py

```diff
@@ -138,3 +138,20 @@ def test_join_then_shuffle(left, right):
     )
 
     assert_gpu_result_equal(q, engine=engine, check_row_order=False)
+
+
+def test_join_conditional(left, right):
```

Comment:
I haven't pulled this down locally to check, but do we have coverage for all 3 cases of 

- left_count < right_count & left_count > 1
- left_count >= right_count & right_count > 1
- the rest

Reply:
We have:
- left_count < right_count & left_count > 1
- left_count > right_count & right_count > 1

We don't technically need any other variations for coverage, but we *can* add them.